xl_auth/permission/views.py

```diff
@@ -76,20 +76,28 @@ def edit(permission_id):
                                permission=permission)
 
 
-@blueprint.route('/delete/<permission_id>', methods=['GET', 'DELETE'])
+@blueprint.route('/delete/<permission_id>', methods=['GET', 'POST'])
 @login_required
 def delete(permission_id):
     """Delete existing permission."""
-    if not current_user.is_admin:
-        abort(403)
-
     permission = Permission.get_by_id(permission_id)
     if not permission:
         flash(_('Permission ID "%(permission_id)s" does not exist', permission_id=permission_id),
               'danger')
-    else:
-        username, collection_code = permission.user.email, permission.collection.code
-        permission.delete()
-        flash(_('Successfully deleted permissions for "%(username)s" on collection "%(code)s".',
-                username=username, code=collection_code), 'success')
-    return redirect(url_for('permission.home'))
+        return redirect(request.args.get('next') or url_for('public.home'))
```

Comment:
Using `request.args.get('next')` this way is dangerous, since we don't check that we redirect the user to a 'safe' place. See [0] for more information.

[0] http://flask.pocoo.org/snippets/62/